### ISSUE TITLE
Do you use the alias files of Wikidata5M?

### ISSUE THREAD

**Issue by lena-schwert** [intfloat/SimKGC#3]
Hello Liang,

I am wondering about your use of the textual descriptions contained in the Wikidata5M dataset. In the paper in chapter 3.2 you write "We first concatenate the textual descriptions of entity h and relation r with a special symbol [SEP] in between". 

But Wikidata5M only contains longer descriptions (extracted from Wikipedia), for entities, not relations. Does this mean that you use the relation alias file as well?

Regarding the entity aliases, I found that the file seems to have data quality issues, which can be seen in the screenshot, e.g. a typo for the Q5 entity. The screenshot is an excerpt from the output of your `preprocess.sh` script.

Seeing that you used the entity alias file at least during processing, could you elaborate how/whether you use it later on, e.g. whether you use it as input to SimKGC?

![image](https://user-images.githubusercontent.com/63639195/163220498-f2455932-898c-4d5f-a34a-8f563e640bd8.png)

**Comment by intfloat** [comment]
Hi,

Thank you for your interest.

Regarding question "Does this mean that you use the relation alias file as well?":
Yes, we use the relation alias file to get the textual description of each relation, as shown in the following function: https://github.com/intfloat/SimKGC/blob/4388ebc0c0011fe333bc5a98d0613ab0d1825ddc/preprocess.py#L188-L198

On the data quality issues, we also found some typos as you mentioned. But the majority of them seems OK based on manual inspection, so we just use the data as it is.

On question "could you elaborate how/whether you use it later on, e.g. whether you use it as input to SimKGC?"
Yes, we use it as input. As described in the paper, we use the  concatenation of entity h and relation r as input. For entity h, we concatenate the entity alias and entity description, which corresponds to the "entity" and "entity_desc" field in your attached figure.
The relevant code snippet is shown here:
https://github.com/intfloat/SimKGC/blob/4388ebc0c0011fe333bc5a98d0613ab0d1825ddc/doc.py#L94-L102
and https://github.com/intfloat/SimKGC/blob/4388ebc0c0011fe333bc5a98d0613ab0d1825ddc/doc.py#L41-L46

Please reach out if I am not clear enough.

Thank you.

**Comment by lena-schwert** [comment]
Thank you very much for your detailed reply!

When I manually checked the entity aliases (some examples below) I found that sometimes the first alias does not even match the meaning of the current Wikidata label. 

But in your case where you concatenate the entity alias (and only max. 10 tokens if I see correctly) and the description I guess that does not necessarily influence the overall information content of the entity text. 

<!DOCTYPE html>

ID | first alias in Wikidata5M | first alias in current Wikidata | label in current Wikidata
-- | -- | -- | --
Q1 | Earth's universe | cosmos | universe
Q2 | mainshock | Planet Earth | Earth
Q3 | Definition of life | biota | life
Q4 | plant death | bereft of life | death
Q5 | Huamn | human being | human
Q76 | bh obama | Barack Hussein Obama II | Barack Obama
Q47551 | ditiano | Titiano da Cadore | Titian
Q471108 | tadashi ōtsuka | Tadashi Ōtsuka | Tadashi Ohtsuka
Q4577869 | 1977 hurricane season | none | 1977 hurricane season
